src/Microsoft.PowerShell.Commands.Diagnostics/GetEventCommand.cs

```diff
@@ -1538,6 +1542,41 @@ private string HandleDataHashValue(Object value)
 
 
         //
+        // HandleNamedDataHashValue helper for hashtable structured query builder.
+        // Constructs and returns named event data field XPath portion as a string.
+        // Fix Issue #2327
+        //
+        private string HandleNamedDataHashValue(String key, Object value)
+        {
+            string ret = "";
+            if (value is Array)
```

Comment:
Good point.  I didn't realize this pattern was used in other places in the code, but I now see it is used in five other places.  This must be old code because it is not a good or common pattern that we use.  I feel it is worth changing in these five places.  Can you make this part of your change?  At the very least I feel that your new change should use the common pattern even if we create a new issue to clean up the rest of the code.